src/include/catalog/catalog_structs.h

```diff
@@ -58,30 +58,47 @@ struct Property : PropertyNameDataType {
 
 struct TableSchema {
 public:
-    TableSchema(string tableName, table_id_t tableID, bool isNodeTable)
-        : tableName{std::move(tableName)}, tableID{tableID}, isNodeTable{isNodeTable} {}
+    TableSchema(string tableName, table_id_t tableID, bool isNodeTable, vector<Property> properties)
+        : tableName{std::move(tableName)}, tableID{tableID}, isNodeTable{isNodeTable},
+          properties{std::move(properties)} {}
 
     virtual ~TableSchema() = default;
 
     static inline bool isReservedPropertyName(const string& propertyName) {
         return propertyName == INTERNAL_ID_SUFFIX;
     }
 
+    inline uint32_t getNumProperties() const { return properties.size(); }
+
+    inline void removeProperty(property_id_t propertyID) {
+        properties.erase(std::remove_if(properties.begin(), properties.end(),
+                             [propertyID](const Property& property) {
+                                 return property.propertyID == propertyID;
+                             }),
+            properties.end());
+    }
+
+    inline bool containProperty(string propertyName) const {
+        return any_of(properties.begin(), properties.end(),
+            [&propertyName](const Property& property) { return property.name == propertyName; });
+    }
+
+    string getPropertyName(property_id_t propertyID) const;
+
 public:
     string tableName;
     table_id_t tableID;
     bool isNodeTable;
+    vector<Property> properties;
```

Comment:
add an issue about replacing isNodeTable boolean field with a table type enum. Enum is always preferred over  boolean check whenever possible.

Reply:
Opened issue: https://github.com/kuzudb/kuzu/issues/1170